docs/decisions/broker-trigger.md

```diff
@@ -0,0 +1,166 @@
+  processing an event. Typically, this is the "sink" in the processing flow of
+  an Event, but could be a data processing system like Spark.
+- **Sink**: a generic term to mean the destination for an event being sent from
+  a component. Synonym for Event Consumer.
+- **Event Source**: a helper object that can be used to connect an Event
+  Producer to the Knative eventing system by routing events to an Event
+  Consumer. E.g. the
+  [GitHub Knative Source](https://github.com/knative/eventing-sources/tree/master/pkg/reconciler/githubsource)
+- **Receive Adapter**: a data plane entity (Knative Service, Deployment, etc)
+  which performs the event routing from outside the Knative cluster to the
+  Knative eventing system. Receive Adapters are created by Event Sources to
+  perform the actual event routing.
+
+## Objects
+
+The following objects are the minimum set needed to experiment and collect
+customer feedback on the Eventing object model, aka an MVP (including rough YAML
+definitions; final naming and field definitions subject to change based on
+implementation experience).
+
+### Broker
+
+A Broker represents a "bucket of events". It is a namespaced object, with
+automation provided to provision a Broker named `default` in
+appropriately-labelled namespaces. It is expected that most users will not need
+to explicitly create a Broker and simply enabling eventing in the namespace to
+create the `default` Broker will be sufficient. (The `default` broker will be
+provisioned automatically by a controller upon appropriate namespace annotation.
+This controller will _reconcile_ on the annotation, and so will recreate the
+`default` Broker if it is deleted. Removing the annotation will be the correct
+way to remove the reconciled Broker.) Assuming that the Broker model is
+successful, future recommendations for Source CRD authors would be to default
+the Source's `spec.sink` attribute to point to the `default` Broker in the
+namespace.
+
+Historical storage and replay of events is not part of the Broker design or MVP.
+
+Broker implements the "Addressable" duck type to allow Source custom resources
+to target the Broker. Broker DNS names should also be predictable (e.g.
+`default-broker` for the `default` Broker), to enable whitebox event producers
+to target the Broker without needing to reconcile against the "Addressable" duck
+type.
+
+#### Broker Object definition
+
+```golang
+type Broker struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec BrokerSpec `json:"spec,omitempty"`
+	Status BrokerStatus `json:"status,omitempty"`
+}
+
+type BrokerSpec struct {
+	// Empty for now
+}
+
+type BrokerStatus struct {
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	Address duckv1alpha1.Addressable `json:"address,omitempty"`
+}
+```
+
+### Trigger
+
+A Trigger represents a registration of interest (request for delivery to an
```

Comment:
The fact that there is a "to"-direction is important in the definition of a Trigger and deserves to be lifted out of the parenthetical.

How about "A Trigger represents a registration of interest in a filtered selection of events (in | routed through | accepted by | handled by) the Broker to be delivered to an Addressable object or URL."?

Reply:
Thanks, done.